app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -52,7 +52,7 @@ def handle_valid_otp
   end
 
   def authentication_context?
-    context == 'authentication'
+    context == 'authentication' || context == 'reauthentication'
```

Comment:
agreed. I was planning on making another pass at this module to see how to improve it.